support/globalconfig/infrastructure.go

```diff
@@ -40,9 +40,10 @@ func ReconcileInfrastructure(infra *configv1.Infrastructure, hcp *hyperv1.Hosted
 	infra.Status.InfrastructureName = hcp.Spec.InfraID
 	infra.Status.ControlPlaneTopology = configv1.ExternalTopologyMode
 	infra.Status.Platform = configv1.PlatformType(platformType)
-	infra.Status.PlatformStatus = &configv1.PlatformStatus{
-		Type: configv1.PlatformType(platformType),
+	if infra.Status.PlatformStatus == nil {
```

Comment:
can clarify why this change is needed?

Reply:
I've added a comment in the PR description.
cc @sjenning if more details are needed.